### PR TITLE
added template to generate new module

### PR DESCRIPTION
**Pull request by jonchenn**
- add CLI template to create a new module
- updated restful_service readme

### CODE REVIEW

**Comment by jonchenn** [comment]
E2E test running: https://github.com/GoogleCloudPlatform/solutions-template/actions/runs/5350244403

**Comment by lavjain** [comment]
Component vs module is confusing:
```
@lavjain ➜ /workspaces/solutions-template (cli-template) $ st component available
Solutions Template (version 1.9.0)

Available modules:

- restful_service
- terraform_gke
- terraform_httplb_cloudrun
```

**Comment by lavjain** [comment]
Help is missing for few of the options:
```
@lavjain ➜ /workspaces/solutions-template (cli-template) $ st --help
Solutions Template (version 1.9.0)

Usage: st [OPTIONS] COMMAND [ARGS]...

  Solutions Template CLI. See
  https://github.com/GoogleCloudPlatform/solutions-template for details.

Options:
  --help  Show this message and exit.

Commands:
  component
  deploy     Build and deploy services.
  destroy    Destory deployment.
  infra
  new        Create a new solution folder.
  tool
  update     Update an existing solution folder.
  version    Print version.
```

**Comment by lavjain** [comment]
IMO, `st component` should bee `st components` (plural)

File: solutions_template/module_template/{{'terraform' if has_terraform_stage}}/stages/{{terraform_stage if has_terraform_stage}}/providers.tf [hunk 1]
```diff
@@ -15,18 +15,18 @@
...
 
+# TODO: Update Terraform providers below.
+
 # Terraform Block
 terraform {
   required_providers {
     google = {
       source  = "hashicorp/google"
-      version = ">= 4.65.2"
+      version = ">= 4.50.0"
```
Comment (lavjain) [inline solutions_template/module_template/{{'terraform' if has_terraform_stage}}/stages/{{terraform_stage if has_terraform_stage}}/providers.tf]: Why downgrading the version?

Comment (jonchenn) [reply solutions_template/module_template/{{'terraform' if has_terraform_stage}}/stages/{{terraform_stage if has_terraform_stage}}/providers.tf]: Good catch. I guess it was from another module. I'll fix this.

**Comment by jonchenn** [comment]
> IMO, `st component` should bee `st components` (plural)

I'm debating whether to use `st modules` vs `st components`. Component here means either the microservice or frontend component, whileas a module may contain component(s) and terraform codes.

Does it make sense or even more confusing?

**Comment by lavjain** [comment]
> > IMO, `st component` should bee `st components` (plural)
> 
> I'm debating whether to use `st modules` vs `st components`. Component here means either the microservice or frontend component, whileas a module may contain component(s) and terraform codes.
> 
> Does it make sense or even more confusing?

For gcloud commands it is plural. I think components makes more sense as it could be frontend or backend. Module would be mainly backend

**Comment by jonchenn** [comment]
Updated CLI with `st components`

**Comment by jonchenn** [comment]
E2E test passed: https://github.com/GoogleCloudPlatform/solutions-template/actions/runs/5385569599/jobs/9774706432

**Review by lavjain — APPROVED** [review]

**Pull request merged**